crates/router/src/core/utils.rs

```diff
@@ -1510,6 +1510,45 @@ pub fn get_incremental_authorization_allowed_value(
     }
 }
 
+pub trait GetPaymentMethodDataForNetworkTransactionId {
```

Comment:
We don't need trait if this function is not polymorphic, instead move this function to impl block of `api_models::mandates::RecurringDetails`